app/views/filters/_form.html.erb

```diff
@@ -62,16 +62,15 @@
 
     <% if show_location_tab? %>
       <div class="tab-pane" id="locations">
-        <% collection = User.current.locations.empty? ? Location : User.current.locations %>
-        <%= multiple_checkboxes f, :locations, f.object, collection, {}, :label => _("Locations") %>
+        <% collection = User.current.locations.empty? ? Location : User.current.my_locations %>
```

Comment:
This allows a non-admin user with no locations assigned to see all locations. This should just be `User.current.my_locations` always.